service/directconnect: EqSe2 location is now EqSe2-EQ or EqSe2-WBE

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Description

We have begun seeing consistent test failures such as:

```
=== CONT  TestAccAWSDxConnectionAssociation_basic
TestAccAWSDxConnectionAssociation_basic: resource_aws_dx_connection_association_test.go:15: After applying this test step, the plan was not empty.
stdout:
An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
-/+ destroy and then create replacement
Terraform will perform the following actions:
# aws_dx_connection_association.test must be replaced
-/+ resource "aws_dx_connection_association" "test" {
connection_id = "dxcon-fh4ovyb8"
~ id            = "dxcon-fh4ovyb8" -> (known after apply)
~ lag_id        = "dxlag-fgz5o5wi" -> (known after apply)
}
# aws_dx_lag.test must be replaced
-/+ resource "aws_dx_lag" "test" {
~ arn                    = "arn:aws:directconnect:us-west-2:*******:dxlag/dxlag-fgz5o5wi" -> (known after apply)
connections_bandwidth  = "1Gbps"
force_destroy          = true
~ has_logical_redundancy = "no" -> (known after apply)
~ id                     = "dxlag-fgz5o5wi" -> (known after apply)
~ jumbo_frame_capable    = true -> (known after apply)
~ location               = "EqSe2-EQ" -> "EqSe2"
name                   = "tf-dx-a6joi"
}
Plan: 2 to add, 0 to change, 2 to destroy.
--- FAIL: TestAccAWSDxConnectionAssociation_basic (22.62s)

=== CONT  TestAccAWSDxConnectionAssociation_multiConns
TestAccAWSDxConnectionAssociation_multiConns: resource_aws_dx_connection_association_test.go:31: After applying this test step, the plan was not empty.
stdout:
An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
-/+ destroy and then create replacement
Terraform will perform the following actions:
# aws_dx_connection_association.test1 must be replaced
-/+ resource "aws_dx_connection_association" "test1" {
connection_id = "dxcon-fhahsuz5"
~ id            = "dxcon-fhahsuz5" -> (known after apply)
~ lag_id        = "dxlag-fg2qk24n" -> (known after apply)
}
# aws_dx_connection_association.test2 must be replaced
-/+ resource "aws_dx_connection_association" "test2" {
connection_id = "dxcon-ffo721s8"
~ id            = "dxcon-ffo721s8" -> (known after apply)
~ lag_id        = "dxlag-fg2qk24n" -> (known after apply)
}
# aws_dx_lag.test must be replaced
-/+ resource "aws_dx_lag" "test" {
~ arn                    = "arn:aws:directconnect:us-west-2:*******:dxlag/dxlag-fg2qk24n" -> (known after apply)
connections_bandwidth  = "1Gbps"
force_destroy          = true
~ has_logical_redundancy = "no" -> (known after apply)
~ id                     = "dxlag-fg2qk24n" -> (known after apply)
~ jumbo_frame_capable    = true -> (known after apply)
~ location               = "EqSe2-EQ" -> "EqSe2"
name                   = "tf-dx-zg9b2"
}
Plan: 3 to add, 0 to change, 3 to destroy.
--- FAIL: TestAccAWSDxConnectionAssociation_multiConns (23.47s)

=== CONT  TestAccAWSDxLag_basic
TestAccAWSDxLag_basic: resource_aws_dx_lag_test.go:19: Check 4/5 error: aws_dx_lag.test: Attribute 'location' expected "EqSe2", got "EqSe2-WBE"
--- FAIL: TestAccAWSDxLag_basic (16.92s)

=== CONT  TestAccAWSDxLag_tags
TestAccAWSDxLag_tags: resource_aws_dx_lag_test.go:58: After applying this test step, the plan was not empty.
stdout:
An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
-/+ destroy and then create replacement
Terraform will perform the following actions:
# aws_dx_lag.test must be replaced
-/+ resource "aws_dx_lag" "test" {
~ arn                    = "arn:aws:directconnect:us-west-2:*******:dxlag/dxlag-fgp7fkdz" -> (known after apply)
connections_bandwidth  = "1Gbps"
force_destroy          = true
~ has_logical_redundancy = "no" -> (known after apply)
~ id                     = "dxlag-fgp7fkdz" -> (known after apply)
~ jumbo_frame_capable    = true -> (known after apply)
~ location               = "EqSe2-WBE" -> "EqSe2"
name                   = "tf-dx-lag-ujibm"
tags                   = {
"Environment" = "production"
"Usage"       = "original"
}
}
Plan: 1 to add, 0 to change, 1 to destroy.
--- FAIL: TestAccAWSDxLag_tags (17.54s)
```

Which seems to suggest that the Equinix Seattle location has been split and can be verified in the Direct Connect API:

```console
$ aws directconnect describe-locations
{
    "locations": [
        {
            "locationCode": "EqSe2-WBE",
            "locationName": "Westin Building Exchange, Seattle, WA | Westin Building Exchange - SEA10 - Main FMMR, Seattle, WA",
            "region": "us-west-2",
            "availablePortSpeeds": [
                "10G"
            ],
            "availableProviders": [
                "Pureport",
                "Beanfield Metroconnect",
                "ePLDT",
                "Datapipe",
                "Telstra",
                "Colt",
                "Verizon",
                "Console Connect",
                "Internet2",
                "Transtelco",
                "Megaport",
                "GTT",
                "Layer42",
                "Equinix, Inc.",
                "Zayo Group",
                "Masergy",
                "QTS",
                "CenturyLink",
                "AT&T",
                "IX Reach",
                "Comcast",
                "Sprint",
                "XO Communications",
                "Telia Carrier",
                "HopOne",
                "PacketFabric",
                "Telus",
                "China Mobile International",
                "Epsilon"
            ]
        },
        {
            "locationCode": "EqSe2-EQ",
            "locationName": "Westin Building Exchange, Seattle, WA | Equinix, SE2, Seattle, WA",
            "region": "us-west-2",
            "availablePortSpeeds": [
                "1G",
                "10G"
            ],
            "availableProviders": [
                "Pureport",
                "Beanfield Metroconnect",
                "ePLDT",
                "Datapipe",
                "Telstra",
                "Colt",
                "Verizon",
                "Console Connect",
                "Internet2",
                "Transtelco",
                "Megaport",
                "GTT",
                "Layer42",
                "Equinix, Inc.",
                "Zayo Group",
                "Masergy",
                "QTS",
                "CenturyLink",
                "AT&T",
                "IX Reach",
                "Comcast",
                "Sprint",
                "XO Communications",
                "Telia Carrier",
                "HopOne",
                "PacketFabric",
                "Telus",
                "China Mobile International",
                "Epsilon"
            ]
        },
```

The testing and documentation needs to be updated to utilize one of the new location codes for consistency. Longer term, should be handled via data source (e.g. #9735) if possible.

### Affected Resource(s)

* aws_dx_connection
* aws_dx_connection_association
* aws_dx_lag

### References

* https://github.com/terraform-providers/terraform-provider-aws/pull/9735

--- a/aws/resource_aws_dx_connection_association_test.go
+++ b/aws/resource_aws_dx_connection_association_test.go
@@ -85,13 +85,13 @@ func testAccDxConnectionAssociationConfig(rName string) string {
 resource "aws_dx_connection" "test" {
   name      = "tf-dx-%s"
   bandwidth = "1Gbps"
-  location  = "EqSe2"
+  location  = "EqSe2-EQ"
 }
 
 resource "aws_dx_lag" "test" {
   name                  = "tf-dx-%s"
   connections_bandwidth = "1Gbps"
-  location              = "EqSe2"
+  location              = "EqSe2-EQ"
   force_destroy         = true
 }
 
@@ -107,19 +107,19 @@ func testAccDxConnectionAssociationConfig_multiConns(rName string) string {
 resource "aws_dx_connection" "test1" {
   name      = "tf-dxconn1-%s"
   bandwidth = "1Gbps"
-  location  = "EqSe2"
+  location  = "EqSe2-EQ"
 }
 
 resource "aws_dx_connection" "test2" {
   name      = "tf-dxconn2-%s"
   bandwidth = "1Gbps"
-  location  = "EqSe2"
+  location  = "EqSe2-EQ"
 }
 
 resource "aws_dx_lag" "test" {
   name                  = "tf-dx-%s"
   connections_bandwidth = "1Gbps"
-  location              = "EqSe2"
+  location              = "EqSe2-EQ"
   force_destroy         = true
 }
 

--- a/aws/resource_aws_dx_connection_test.go
+++ b/aws/resource_aws_dx_connection_test.go
@@ -26,7 +26,7 @@ func TestAccAWSDxConnection_basic(t *testing.T) {
 					testAccCheckAwsDxConnectionExists(resourceName),
 					resource.TestCheckResourceAttr(resourceName, "name", connectionName),
 					resource.TestCheckResourceAttr(resourceName, "bandwidth", "1Gbps"),
-					resource.TestCheckResourceAttr(resourceName, "location", "EqSe2"),
+					resource.TestCheckResourceAttr(resourceName, "location", "EqSe2-EQ"),
 					resource.TestCheckResourceAttr(resourceName, "tags.%", "0"),
 				),
 			},
@@ -116,7 +116,7 @@ func testAccDxConnectionConfig(n string) string {
 resource "aws_dx_connection" "test" {
   name      = "%s"
   bandwidth = "1Gbps"
-  location  = "EqSe2"
+  location  = "EqSe2-EQ"
 }
 `, n)
 }
@@ -126,7 +126,7 @@ func testAccDxConnectionConfig_tags(n string) string {
 resource "aws_dx_connection" "test" {
   name      = "%s"
   bandwidth = "1Gbps"
-  location  = "EqSe2"
+  location  = "EqSe2-EQ"
 
   tags = {
     Environment = "production"
@@ -141,7 +141,7 @@ func testAccDxConnectionConfig_tagsChanged(n string) string {
 resource "aws_dx_connection" "test" {
   name      = "%s"
   bandwidth = "1Gbps"
-  location  = "EqSe2"
+  location  = "EqSe2-EQ"
 
   tags = {
     Usage = "changed"

--- a/aws/resource_aws_dx_lag_test.go
+++ b/aws/resource_aws_dx_lag_test.go
@@ -27,7 +27,7 @@ func TestAccAWSDxLag_basic(t *testing.T) {
 					testAccCheckAwsDxLagExists(resourceName),
 					resource.TestCheckResourceAttr(resourceName, "name", lagName1),
 					resource.TestCheckResourceAttr(resourceName, "connections_bandwidth", "1Gbps"),
-					resource.TestCheckResourceAttr(resourceName, "location", "EqSe2"),
+					resource.TestCheckResourceAttr(resourceName, "location", "EqSe2-EQ"),
 					resource.TestCheckResourceAttr(resourceName, "tags.%", "0"),
 				),
 			},
@@ -43,7 +43,7 @@ func TestAccAWSDxLag_basic(t *testing.T) {
 					testAccCheckAwsDxLagExists(resourceName),
 					resource.TestCheckResourceAttr(resourceName, "name", lagName2),
 					resource.TestCheckResourceAttr(resourceName, "connections_bandwidth", "1Gbps"),
-					resource.TestCheckResourceAttr(resourceName, "location", "EqSe2"),
+					resource.TestCheckResourceAttr(resourceName, "location", "EqSe2-EQ"),
 					resource.TestCheckResourceAttr(resourceName, "tags.%", "0"),
 				),
 			},
@@ -138,7 +138,7 @@ func testAccDxLagConfig(n string) string {
 resource "aws_dx_lag" "test" {
   name                  = "%s"
   connections_bandwidth = "1Gbps"
-  location              = "EqSe2"
+  location              = "EqSe2-EQ"
   force_destroy         = true
 }
 `, n)
@@ -149,7 +149,7 @@ func testAccDxLagConfig_tags(n string) string {
 resource "aws_dx_lag" "test" {
   name                  = "%s"
   connections_bandwidth = "1Gbps"
-  location              = "EqSe2"
+  location              = "EqSe2-EQ"
   force_destroy         = true
 
   tags = {
@@ -165,7 +165,7 @@ func testAccDxLagConfig_tagsChanged(n string) string {
 resource "aws_dx_lag" "test" {
   name                  = "%s"
   connections_bandwidth = "1Gbps"
-  location              = "EqSe2"
+  location              = "EqSe2-EQ"
   force_destroy         = true
 
   tags = {

--- a/website/docs/guides/version-2-upgrade.html.md
+++ b/website/docs/guides/version-2-upgrade.html.md
@@ -536,7 +536,7 @@ For example, given this previous configuration:
 resource "aws_dx_lag" "example" {
   name                  = "example"
   connections_bandwidth = "1Gbps"
-  location              = "EqSe2"
+  location              = "EqSe2-EQ"
   number_of_connections = 1
 }
 ```
@@ -547,13 +547,13 @@ An updated configuration:
 resource "aws_dx_connection" "example" {
   name      = "example"
   bandwidth = "1Gbps"
-  location  = "EqSe2"
+  location  = "EqSe2-EQ"
 }
 
 resource "aws_dx_lag" "example" {
   name                  = "example"
   connections_bandwidth = "1Gbps"
-  location              = "EqSe2"
+  location              = "EqSe2-EQ"
 }
 
 resource "aws_dx_connection_association" "example" {

--- a/website/docs/r/dx_connection_association.html.markdown
+++ b/website/docs/r/dx_connection_association.html.markdown
@@ -16,13 +16,13 @@ Associates a Direct Connect Connection with a LAG.
 resource "aws_dx_connection" "example" {
   name      = "example"
   bandwidth = "1Gbps"
-  location  = "EqSe2"
+  location  = "EqSe2-EQ"
 }
 
 resource "aws_dx_lag" "example" {
   name                  = "example"
   connections_bandwidth = "1Gbps"
-  location              = "EqSe2"
+  location              = "EqSe2-EQ"
 }
 
 resource "aws_dx_connection_association" "example" {

